src/features/utils/instrument-base.js

```diff
@@ -56,6 +61,20 @@ export class InstrumentBase extends FeatureBase {
    */
   importAggregator (argsObjFromInstrument = {}) {
     if (this.featAggregate || !this.auto) return
+
+    if (this.requiresOptIn && !this.optedIn) {
+      // this feature requires an opt in...
+      // wait for API to be called
+      this.ee.on(`${this.featureName}-opt-in`, () => {
+        // register the feature to drain only once the API has been called, it will drain when importAggregator finishes for all the features
+        // called by the api in that cycle
+        registerDrain(this.agentIdentifier, this.featureName)
+        this.optedIn = true
+        this.importAggregator()
```

Comment:
The way this is done makes it non-obvious for any edge case bugs, such as similar to the interaction-timer issue with too many redundant handlers over time.

I recommend getting rid of `this.optedIn` (less unneeded logic vars the better) and creating a separate function rather than recursing itself. If you want to doubly ensure the code path of `this.importAggregator()` only runs once, you can either make the ee handler a singly function or tbh `featAggregate` already does that job, though you could also just set `this.auto = false` for extra assurance.

Reply:
if youre worried about extra logic, all of that is equally as complicated as just staging another call to importAggregator, and there is no way for it to trigger ee.on more than once, so thats not really an issue here...

but i do like the this.auto = false route so i will modify to use that